src/sema/mod.rs

```diff
@@ -39,9 +41,16 @@ pub const SOLANA_BUCKET_SIZE: u64 = 251;
 pub const SOLANA_FIRST_OFFSET: u64 = 16;
 pub const SOLANA_SPARSE_ARRAY_SIZE: u64 = 1024;
 
+/// Performs semantic analysis and checks for unused variables
```

Comment:
The comment is not really correct. This function is the entry point for sema. Maybe move the comment from sema_file to here and put this comment inside the function.

Reply:
I moved the comments and added a new one to `sema_file`